### PR TITLE
Fix add node type bugs where resource group had other resources that were not part of the cluster

### PR DESCRIPTION
**Pull request by a-santamaria**
## Description

Fix add node type cmdlet bugs
    - NullReferenceException bug when resource group had other vmss not related to the service fabric cluster. Fixes issue: https://github.com/Azure/azure-powershell/issues/8681
    - Fix bug where cmdlet failed if virtualNetwork was in a different resource group that the cluster. fixes issue: https://github.com/Azure/azure-powershell/issues/8407

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by a-santamaria** [comment]
@markcowl could you please take a look at this pr?

File: src/ServiceFabric/ServiceFabric/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Fix add node type cmdlet bugs:
```
Comment (markcowl) [inline src/ServiceFabric/ServiceFabric/ChangeLog.md]: Can we add any test validation / regression protection for this fix to the ServiceFabric tests?

Comment (a-santamaria) [reply src/ServiceFabric/ServiceFabric/ChangeLog.md]: sure, I'll add the test

Comment (a-santamaria) [reply src/ServiceFabric/ServiceFabric/ChangeLog.md]: I recorded the tests again adding the necessary things so the fixes I made were validated

**Comment by cormacpayne** [comment]
@a-santamaria the CredScan tool picked up a couple of issues in this PR:

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=75843&view=artifacts

Would you mind taking a look at these and suppressing the exceptions if these are dummy values and/or the corresponding test resource has been deleted?

**Pull request merged**